framework/src/controller/controller.js

```diff
@@ -147,15 +151,16 @@ class Controller {
 
 	async _loadModules(modules) {
 		// To perform operations in sequence and not using bluebird
-
 		// eslint-disable-next-line no-restricted-syntax
 		for (const alias of Object.keys(modules)) {
-			// eslint-disable-next-line no-await-in-loop
-			await this._loadInMemoryModule(
-				alias,
-				modules[alias].klass,
-				modules[alias].options
-			);
+			const { klass, options } = modules[alias];
+			if (options.useSocketChannel) {
+				// eslint-disable-next-line no-await-in-loop
+				await this._loadModuleWithSocketChannel(alias, klass, options);
```

Comment:
I think `_loadModuleAsAChildProcess` or `_loadModuleAsASepareteProcess` would be clearer of what this method is doing (if I am not mistaken). Let me know what you think on this :)

Reply:
This will be renamed in https://github.com/LiskHQ/lisk/issues/3059
For now there is no child process and this method is loading `httpApi` module using the `ChildProcessChannel` but not as a child process itself.